### PR TITLE
Wk/types 001 - Vouchers (ListGiftCardTransactions, ExportGiftCardTransactions)

### PR DESCRIPTION
**Pull request by weronika-kurczyna**
**New exported types/interfaces**
- ListGiftCardTransactionsResponse
- ListGiftCardTransactionsType
- ListGiftCardTransactionsTypeDataObject
- ExportGiftCardTransactionsBody
- ExportGiftCardTransactionsResponse
- ExportGiftCardTransactionsParams
- ExportGiftCardTransactionsFields
- ExportGiftCardTransactionsParamsFilters

Added support for following endpoints:

- vouchers
    - GET /vouchers/{code}/transactions
    - POST /vouchers/{code}/transactions/export
    
Note: ListGiftCardTransactions response includes also `total` property (which is only visible when there are not transactions). This property was previously omitted in openApi file.

### CODE REVIEW

File: packages/sdk/src/types/Vouchers.ts [hunk 1]
```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
...
+			id: string
+		}
+	}
+	related_transaction_id: string | null
+	created_at: string
+}
+
+export interface ExportGiftCardTransactionsBody {
+	parameters?: {
+		order?: '-created_at' | 'created_at'
```
Comment (p-zielinski) [inline packages/sdk/src/types/Vouchers.ts:312]: this shall be equal to ExportGiftCardTransactionsFields and ExportGiftCardTransactionsFields.map(e => `-${e}`)

File: packages/sdk/src/types/Vouchers.ts [hunk 2]
```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
...
+
+export interface ListGiftCardTransactionsQuery {
+	limit?: number
+	page?: number
+}
+
+export interface ListGiftCardTransactionsResponse {
+	object: 'list'
+	data_ref: 'data'
+	data: ListGiftCardTransactionsTypeDataObject[] | []
```
Comment (p-zielinski) [inline packages/sdk/src/types/Vouchers.ts:263]: simply `ListGiftCardTransactionsTypeDataObject[]` ?

File: packages/sdk/src/types/Vouchers.ts [hunk 3]
```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
...
+	| 'CREDITS_REFUND'
+	| 'CREDITS_ADDITION'
+	| 'CREDITS_REMOVAL'
+
+export interface ListGiftCardTransactionsTypeDataObject {
+	id: string
+	source_id: string | null
+	voucher_id: string
+	campaign_id: string | null
+	source: 'voucherify-web-ui' | 'API' | null
```
Comment (p-zielinski) [inline packages/sdk/src/types/Vouchers.ts:279]: are you sure about that?

File: packages/sdk/src/Vouchers.ts [hunk 4]
```diff
@@ -89,4 +89,16 @@ export class Vouchers {
...
+	 * @see https://docs.voucherify.io/reference/list-gift-card-transactions
+	 */
+	public listGiftCardTransactions(code: string, query?: T.ListGiftCardTransactionsQuery) {
+		return this.client.get<T.ListGiftCardTransactionsResponse>(`/vouchers/${encode(code)}/transactions`, query)
+	}
+	/**
+	 * @see https://docs.voucherify.io/reference/export-gift-card-transactions
+	 */
+	public exportGiftCardTransactions(code: string, body: T.ExportGiftCardTransactionsBody) {
+		return this.client.post<T.ExportGiftCardTransactionsResponse>(`/vouchers/${encode(code)}/transactions/export`, body)
```
Comment (p-zielinski) [inline packages/sdk/src/Vouchers.ts:102]: just to be sure?
Is there a case when you will get
```
{
...
status: DONE
result: string
}
```
here?

Comment (weronika-kurczyna) [reply packages/sdk/src/Vouchers.ts:102]: it looks like it's an endpoint dedicated only to schedule the export and it always returns "status": "SCHEDULED", need to check this

File: packages/sdk/src/types/Vouchers.ts (hunk 3 above)
Comment (weronika-kurczyna) [reply packages/sdk/src/types/Vouchers.ts:279]: I received a response in which the `source` is `null`
The `id` is always in the response (id of the transaction)
`source_id` is not defined in many cases, but it's returned in every response
`voucher_id` must be defined
if it's a standalone the `campaign_id` could be null

Comment (weronika-kurczyna) [reply packages/sdk/src/types/Vouchers.ts:279]: or perhaps you are referring to some specific property? :)

File: packages/sdk/src/types/Vouchers.ts (hunk 1 above)
Comment (weronika-kurczyna) [reply packages/sdk/src/types/Vouchers.ts:312]: do you mean that each field in ExportGiftCardTransactionsFields can be an `order` value (descending or ascending)?

Comment (weronika-kurczyna) [reply packages/sdk/src/types/Vouchers.ts:312]: even if you call the export from the V% dashboard the only possible options are '-created_at' | 'created_at'

**Comment by weronika-kurczyna** [comment]
@p-zielinski I hope the changes will be correct, I tried by best, If you have any comments, I will change it ;)

**Pull request closed without merging**